### PR TITLE
feat: Add Validate Verifiable Presentation when provider request

### LINKED ISSUE #80: Add validate `VerifiablePresentation`.
## Background
The oracle is validate a JSON schema for now, but we are planning to validate a VP(Verifiable Presentation).
So, We need to implement validating the VP(Verifiable Presentation) which has a VC(Verifiable Credential) and `Proofs`.

## Implement
When the provider request the validate the data which includes a VP, the oracle will decrypt the data and validate it.
The VP and the proof will be validate and after validation is success, the data will be encrypted by `secretKey` and return as `DataCertificate`. Also, in near feature the oracle will be validate the not only VP and the proof, but also `PresentationDefinition`. The VP validation will use a [medibloc/vc-sdk](https://github.com/medibloc/vc-sdk) which applying aries-framework-go v0.1.8

### PR DESCRIPTION
**Pull request by inchori**
Closes #80

### CODE REVIEW

File: validation/verifiable_presentation.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	if err != nil {
+		return err
+	}
+
+	// TODO: validate VP with the PresentationDefinition
+
+	return nil
+}
+
+func validateVPProofs(proofs *vc.ProofIterator) error {
```
Comment (inchori) [inline validation/verifiable_presentation.go:35]: I think that need to find how to validate VP proofs more.

**Comment by youngjoon-lee** [comment]
Let's postpone merging this PR until PR #79 is merged.

**Comment by H4NLee** [comment]
I opened a new PR for improvement of VP verification (#91).
Please review the PR :)

**Comment by inchori** [comment]
When the #91 merged into this branch, I'll modify some code and re-request the review.

**Review by H4NLee — APPROVED** [review]
lgtm

**Review by audtlr24 — APPROVED** [review]
lgtm

File: server/rpc/interceptor/auth/auth_test.go [hunk 2]
```diff
@@ -6,6 +6,8 @@ import (
 	"testing"
```
Comment (gyuguen) [inline server/rpc/interceptor/auth/auth_test.go:6]: I realize that there's no need to create a new mockQueryClient struct as it already exists in `mocks/query_client.go.` 
Therefore, I will delete this unnecessary struct soon."

File: validation/verifiable_presentation_test.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		Challenge:          "this is a challenge",
+		Created:            "2017-06-18T21:19:10Z",
+	})
+	require.NoError(t, err)
+
+	err = ValidateVP(mockVDR, vpBytes, nil)
+	require.NoError(t, err)
+}
+
+type MockVDR struct {
```
Comment (gyuguen) [inline validation/verifiable_presentation_test.go:80]: If the `MockVDR` is used in another test, it can be moved to the mocks package.

**Review by gyuguen — APPROVED** [review]
LGTM

File: validation/verifiable_presentation_test.go (hunk 3 above)
Comment (H4NLee) [reply validation/verifiable_presentation_test.go:80]: I'll handle this mock VDR things. Thank you for your comments

File: validation/verifiable_presentation.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+
+	"github.com/medibloc/vc-sdk/pkg/vc"
+)
+
+type didResolver interface {
+	Resolve(did string, opts ...vdr.DIDMethodOption) (*did.DocResolution, error)
+}
```
Comment (H4NLee) [inline validation/verifiable_presentation.go:15]: I guess [gobind doesn't support this kinds of export of interface](https://pkg.go.dev/golang.org/x/mobile/cmd/gobind#hdr-Type_restrictions), so I had to made this private in `vc-sdk`.
(When I make this public, [`build-android` fails](https://github.com/medibloc/vc-sdk/actions/runs/4288965665/jobs/7471532744)..
Thus, I make this interface here too.. I don't like to define this duplicated interface. Any good idea? :sob:

File: validation/verifiable_presentation_test.go (hunk 3 above)
Comment (H4NLee) [reply validation/verifiable_presentation_test.go]: done in [db60d2b](https://github.com/medibloc/panacea-oracle/pull/84/commits/db60d2b1096424845df3f8df74b0210c003dc5ab).
Additionally, we may need more tests for VC/VP validation in oracle. I'll cover it in another PR.

File: validation/verifiable_presentation.go (hunk 4 above)
Comment (youngjoon-lee) [reply validation/verifiable_presentation.go:15]: No worries about gobind for now. For mobile binding, I think we need to refactor the entire structure.

**Review by youngjoon-lee — APPROVED** [review]
lgtm

**Pull request merged**